config/v1/types_image.go

```diff
@@ -67,6 +84,14 @@ type ImageSpec struct {
 	// internal cluster registry.
 	// +optional
 	RegistrySources RegistrySources `json:"registrySources"`
+
+	// imageStreamImportMode controls the import mode behaviour of imagestreams.
+	// It can be set to `Legacy` or `PreserveOriginal`. If this value is specified,
+	// this setting is applied to all newly created imagestreams which do not have the
+	// value set.
+	// +openshift:enable:FeatureGate=ImageStreamImportMode
+	// +optional
+	ImageStreamImportMode ImportModeType `json:"imageStreamImportMode,omitempty"`
```

Comment:
This is a configuration API, so we prefer to drop the omitempty, and allow `""` as an option for the enum.

That allows the rendered resource to show the field name and allows users to discover that this is a configurable option

Reply:
I did see that in some of the other enums..Does it mean it will always have a default value and be present in the spec? or does it mean configuring it empty is allowed?

Reply:
added the `""` enum option.

Reply:
It means that by default it will show up as `imageStreamImportMode: ""`, which will show users that it is an option they could choose to configure. You just need to document what happens when it is the empty string